pygeos/linear.py

```diff
@@ -140,3 +146,25 @@ def shared_paths(a, b, **kwargs):
     <pygeos.Geometry GEOMETRYCOLLECTION (MULTILINESTRING EMPTY, MULTILINESTRING ...>
     """
     return lib.shared_paths(a, b, **kwargs)
+
+
+def shortest_line(a, b, **kwargs):
+    """
+    Returns the shortest line between geom1 and geom2.
+
```

Comment:
It would likely be helpful to clarify if the endpoints of the lines created here (output of `GEOSNearestPoints_r`) are interpolated on the nearest lines between the two geometries, if the inputs are lines or polygons.  At least, that's my interpretation of the dostring for [JTS](https://locationtech.github.io/jts/javadoc/org/locationtech/jts/operation/distance/DistanceOp.html)
> The distance computation also finds a pair of points in the input geometries which have the minimum distance between them. If a point lies in the interior of a line segment, the coordinate computed is a close approximation to the exact point.

As opposed to finding the nearest _existing_ vertices on each geometry that are the closest of other existing vertices between the geometries.

Reply:
I am not fully sure how to interpret "the coordinate computed is a close approximation to the exact point". 

Would be it be sufficient to indicate that the resulting endpoints aren't necessarily existing vertices, but can be any point along the line segment? (that seems a simpler version as saying that the points "are interpolated on the nearest lines")

Reply:
> to meant that there exists a theoretical exact nearest point within the nearest line segment facing the other geometry, and that the method should get very close but not necessarily exactly equal to that theoretical point.

What I am wondering is whether this "not necessarily exactly equal" is just due to inherent floating point precision limitations, or if there is something more specific to this algorithm (as the floating point precision of coordinates is something that can come up in many GEOS operations)